### PR TITLE
contrib: Replace developer keys with list of pgp fingerprints

### PR DESCRIPTION
**Pull request by maflcko**
Having to host a copy of the keys in this repo was a common source of discussion and distraction, caused by problems such as:

* Outdated keys. Unclear whether and when to replace by fresh copies.
* Unclear when to add a key of a new developer or Gitian builder.

The problems are solved by
* Having no keys but only the fingerprints
* Adding a rule of thumb, when to add a new key

<strike>Moving the keys to a different repo solves none of these issues, but since the keys are not bound to releases or git branches of Bitcoin Core, they should live somewhere else.

Obviously, all keys are hosted and distributed on key servers, but were added to the repo solely for convenience and redundancy.

Moving the mirror of those keys to a different repo makes it less distracting to update them -- let's say -- prior to every major release.

I updated our `doc/release-process.md` to reflect the new location.

DEPENDS_ON https://github.com/bitcoin-core/gitian.sigs/pull/621
</strike>

### CODE REVIEW

**Comment by jonasschnelli** [comment]
Moving the keys away from this repository would loose the potential for a "verification-chain" (signed builds contains keys to verify future releaes).
Shipping the keys in a verifiable binary (or now via the tar ball) adds additional security.

My long term use case is to have a binary verifier shipped with bitcoin core (same signing and verification process).
Therefore we could add keys (probably new secp256k1 keys) to the binary and allow verification of future releases.

Maybe we could do that once we have that use case more sketched out.

**Comment by MeshCollider** [comment]
@laanwj suggested leaving the key fingerprints in the main repo (https://github.com/bitcoin/bitcoin/pull/11845#issuecomment-351387138)

**Comment by laanwj** [comment]
Concept ACK.

Having the keys in a separate repository also means that we can keep the keys up to date there - no problem with commit noise, no more need for pulling from gpg keyservers at the beginning of the travis build.

> @laanwj suggested leaving the key fingerprints in the main repo

Yes, having the key fingerprints in the main repo is enough to verify the keys are correct.

File: contrib/gitian-keys/README.md [hunk 1]
```diff
@@ -1,16 +1,55 @@
...
 ```sh
-gpg --import ./*.pgp
 gpg --refresh-keys
 ```
+
+To fetch keys of Gitian builders and active developers, feed the list of
+fingerprints of the primary keys into gpg:
+
+```sh
+echo "
```
Comment (laanwj) [inline contrib/gitian-keys/README.md]: Maybe add a separate script? Such a large snippet is hard to copy/paste.

**Comment by luke-jr** [comment]
Would be nice to have names with the fingerprints.

Maybe:

```
...
AEC1884398647C47413C1C3FB1179EB7347DC10D John Doe
" | while read fingerprint keyholder_name; do
...
```

**Comment by MeshCollider** [comment]
Concept ACK

**Comment by jonasschnelli** [comment]
> Having the keys in a separate repository also means that we can keep the keys up to date there - no problem with commit noise, no more need for refreshing keys from gpg keyservers at the beginning of the travis build (which fails half the time).

Good point.

Concept ACK

File: contrib/gitian-keys/README.md [hunk 2]
```diff
@@ -1,16 +1,26 @@
-PGP keys
-========
+## PGP keys of Gitian builders and Developers
 
-This folder contains the public keys of developers and active contributors.
+This list contains the public keys of Gitian builers and active developers.
```
Comment (achow101) [inline contrib/gitian-keys/README.md]: s/builers/builders

**Review by achow101 — COMMENTED** [review]
utACK fa38d529a75a6cd1a51326c05170b3db2606fd14 modulo typo

Maybe this should include the fingerprint of the release key too?

It might also be better to move this to a more generic pgp keys folder instead of specifically `gitian-keys`.

**Comment by MarcoFalke** [comment]
Fixed @achow101 nit and fixed up, should be easy to re-ACK.

> Maybe this should include the fingerprint of the release key too?

Happy to do that, but I think this is separate from the current goal. Will add a commit on top if others agree on that.

> It might also be better to move this to a more generic pgp keys folder instead of specifically gitian-keys.

Note that the fingerprints of the maintenance keys are listed in https://github.com/bitcoin/bitcoin/blob/62fdf9b07087b80d2142799bdd2324f61483359d/contrib/verify-commits/trusted-keys and the ones in this folder are in fact only used for gitian signatures. So I guess the name is fine to keep for now.

File: contrib/gitian-keys/README.md [hunk 3]
```diff
@@ -1,16 +1,26 @@
-PGP keys
-========
+## PGP keys of Gitian builders and Developers
 
-This folder contains the public keys of developers and active contributors.
+This list contains the public keys of Gitian builders and active developers.
```
Comment (fanquake) [inline contrib/gitian-keys/README.md]: nit: "keys.txt contains"

File: contrib/gitian-keys/keys.txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+912FD3228387123DC97E0E57D5566241A0295FA9 BtcDrak
+C519EBCF3B926298946783EFF6430754120EC2F4 Christian Decker (cdecker)
+F20F56EF6A067F70E8A5C99FFF95FAA971697405 centaur
+C060A6635913D98A3587D7DB1C2491FFEB0EF770 Cory Fields
+BF6273FAEF7CC0BA1F562E50989F6B3048A116B5 Dev Random
+9A1689B60D1B3CCE9262307A2F40A9BF167FBA47 Erik Mossberg (erkmos)
+E777299FC265DD04793070EB944D35F9AC3DB76A Michael Ford
+01CDF4627A3B88AAE4A571C87588242FBE38D3A8 Gavin Andresen
+D3CC177286005BB8FF673294C5242A1AB3936517 jl2012
+D2D1085B9425F9DEFA06E7932270E30C522739F6 Jonas Schnelli
```
Comment (fanquake) [inline contrib/gitian-keys/keys.txt]: This key "D2D1085B9425F9DEFA06E7932270E30C522739F6" has expired, and I couldn't look it up. We could probably remove it given @jonasschnelli has another key listed.

File: contrib/gitian-keys/keys.txt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+CA03882CB1FC067B5D3ACFE4D300116E1C875A3D MeshCollider
+9692B91BBF0E8D34DFD33B1882C5C009628ECF0C Michagogo
+37EC7D7B0A217CDB4B4E007E7FAB114267E4FA04 Peter Todd
+D62A803E27E7F43486035ADBBCD04D8E9CCCAC2A Paul Rabahy
+D762373D24904A3E42F33B08B9A408E71DAAC974 Pieter Wuille (Location: Leuven, Belgium)
+133EAC179436F14A5CF1B794860FEB804E669320 Pieter Wuille
+ED9BDF7AD6A55E232E84524257FF9BDBCC301009 Sjors Provoost
+77E72E69DA7EE0A148C06B21B34821D4944DE5F7 Nils Schneider
+79D00BAC68B56D422F945A8F8E3A8F3247DBCBBF Willy Ko
+AEC1884398647C47413C1C3FB1179EB7347DC10D Warren Togami
```
Comment (fanquake) [inline contrib/gitian-keys/keys.txt]: I verified that I was able to retrieve all the other keys, and that the names listed here were correct.

**Comment by practicalswift** [comment]
Concept ACK

File: contrib/gitian-keys/keys.txt [hunk 6]
```diff
@@ -0,0 +1,27 @@
+152812300785C96444D3334D17565732E08E5E41 Andrew Chow
```
Comment (akx20000a) [inline contrib/gitian-keys/keys.txt]: Could you add my key fingerprints?
617C90010B3BD370B0AC7D424BB42E31C79111B8 Akira Takizawa

**Review by ghost — CHANGES_REQUESTED** [review]

File: contrib/gitian-keys/keys.txt (hunk 6 above)
Comment (MarcoFalke) [reply contrib/gitian-keys/keys.txt]: Removal and addition of keys will happen in separate pull requests. You should wait until this one is merged.  
Might want to help by review, though.

**Comment by laanwj** [comment]
utACK fabb72b

File: contrib/gitian-keys/keys.txt (hunk 6 above)
Comment (laanwj) [reply contrib/gitian-keys/keys.txt]: @akx20000a This has been merged, you can submit your PR now

**Pull request merged**